matter_server/server/server.py

```diff
@@ -166,6 +169,34 @@ def signal_event(self, evt: EventType, data: Any = None) -> None:
             else:
                 callback(evt, data)
 
+    def scope_ipv6_lla(self, ip_addr: str) -> str:
+        """Scope IPv6 link-local addresses to primary interface.
+
+        IPv6 link-local addresses received through the websocket might have no
+        scope_id or a scope_id which isn't valid on this device. Just assume the
+        device is connected on the primary interface.
+        """
+        ip_addr_parsed = ipaddress.ip_address(ip_addr)
+        if not ip_addr_parsed.is_link_local or ip_addr_parsed.version != 6:
+            return ip_addr
+
+        ip_addr_parsed = cast(ipaddress.IPv6Address, ip_addr_parsed)
+
+        if ip_addr_parsed.scope_id is not None:
+            # This type of IPv6 manipulation is not supported by the ipaddress lib
+            ip_addr = ip_addr.split("%")[0]
+
+        # Rely on host OS routing table
+        if self.primary_interface is None:
+            return ip_addr
+
+        self.logger.debug(
+            "Setting scope of link-local IP address %s to %s",
+            ip_addr,
+            self.primary_interface,
+        )
+        return ip_addr + "%" + self.primary_interface
```

Comment:
Please use an f-string instead of string concatenation.

Reply:
Addressed with #504.